src/backend/context.rs

```diff
@@ -700,6 +733,19 @@ impl Context {
 			Event::WindowEvent(WindowEvent::Resized(event)) => {
 				let _ = self.resize_window(event.window_id, event.size);
 			},
+			Event::WindowEvent(WindowEvent::MouseWheel(event)) => {
+				let delta = match event.delta {
+					winit::event::MouseScrollDelta::LineDelta(_, y) => y,
+					_ => 0.0
+				};
+				let _ = self.zoom_window(event.window_id, delta * 0.01);
+			},
+			Event::WindowEvent(WindowEvent::MouseMove(event)) => {
+				if event.buttons.is_pressed(event::MouseButton::Left) {
+					let direction = self.mouse_cache.get_direction(event.window_id, event.device_id).unwrap_or_else(|| [0.0, 0.0].into());
```

Comment:
The applied translation should match exactly with the mouse movement, so cursor *direction* isn't enough. It should look like you're dragging the image along with the mouse movement. So it needs to take into account current zoom level and window size.